internal/credentials/credentials.go

```diff
@@ -43,10 +41,22 @@ type Credentials struct {
 	SSHPrivateKey string
 }
 
-type Helper func(
-	ctx context.Context,
-	project string,
-	credType Type,
-	repoURL string,
-	secret *corev1.Secret,
-) (*Credentials, error)
+// Provider is an interface for providing credentials for a given type,
+// repository URL and data values.
+type Provider interface {
+	// Supports returns true if the provider can provide credentials for the
+	// given type, repository URL and data values. Otherwise, it should return
+	// false.
+	Supports(credType Type, repoURL string, data map[string][]byte) bool
```

Comment:
All the providers seem to invoke `Supports()` themselves and I don't see anywhere where it's invoked from elsewhere. I'm in agreement with the notion of separating the logic for whether a provider is supported from the actually credential retrieval logic, but I am wondering if this isn't more of an implementation detail that doesn't need to be exposed as part of the `Provider` interface.

Reply:
The reason I exported it is because I can think of scenarios in which you may want to put together a set of providers by testing their support without actually wanting to retrieve credentials at that moment.

Reply:
sgtm